### ISSUE TITLE
SRC indices do not have confidence bounds

### ISSUE THREAD

**Issue by mbaudin47** [openturns/openturns#1056]
When we estimate SRC indices, we cannot get neither confidence bounds nor the distribution. 

There are several consequences.

- When the number of simulations is set, we cannot know the precision of the SRC indices after the simulation was done. This is unfortunate, since we might take decisions (e.g. set a random variable to its mean, given that its SRC indice is close to zero) which might be unreliable given the precision of the estimate. 
- It is impossible to create an incremental algorithm which estimates the SRC indices. Indeed, if we do not know the precision of the estimate, we cannot create an algorithm which estimates SRC indices given a precision given by the user.

This sounds weird to me, given that OT already provides confidence bounds for Sobol' indices, which seems far more complicated at first. 

The simplest option is to use bootstrap in order to get confidence bounds. The next action would be to create the incremental algorithm.

I have a Python prototype, with figures: I attach it in the ticket for further discussion.

**Comment by mbaudin47** [comment]
The following is a draft. 

[casCrue-4vars-SRC-with-precision.txt](https://github.com/openturns/openturns/files/3007638/casCrue-4vars-SRC-with-precision.txt)

This uses the flooding test case and uses the CorrelationAnalysis.SRC function to estimate the indices. These are based on a plain Monte-Carlo sampling with size 100 (this value is low, because the flooding model is almost linear). The bootstrap sampling size is 100 and I used 95% confidence bounds.

A side note: with PR #904, the estimate will be better, with the possibility of detecting an inaccurate linear model. With PR #900, the graphics of SRC indices without bounds will be straightforward with OT 1.13.

The following is the graphics without bounds.

![SRC-without-bounds](https://user-images.githubusercontent.com/31351465/54990049-18cb5180-4fba-11e9-87fc-75af8b081a28.png)

The following is the graphics with 95% bounds.

![SRC-with-bounds](https://user-images.githubusercontent.com/31351465/54990058-1ec13280-4fba-11e9-8d35-87a1f13f6322.png)

**Comment by sofianehaddad** [comment]
@mbaudin47 we can get in mind Bootstrap or other alternatives (similar to the ones implemented for Sobol)

Concerning the incremental algo, It seems more complicated.  Sobol indices are **independent** of the model behaviour (we should get the existence of second order indices) wheras **SRC** have interest if linearity!

**Comment by mbaudin47** [comment]
@sofianehaddad : you are right - if, during the execution of the incremental algorithm, this reveals that the model is not linear, the basic hypothesis is not satisfied. What should be the behavior in this case? May we stop the algorithm? This would make sense. The criteria may be based on the sum of the (unscaled) SRC indices: if this falls below a given threshold (say 0.9 for instance), then the algorithm should stop. This, however, may prevent the algorithm in the early phases where the input sample is too small. Hence, the block size should be large enough to pass through this.

**Comment by jschueller** [comment]
do we have the paper ? I cannot find it anymore

**Comment by mbaudin47** [comment]
Oups, I missed your message: which paper are you interested in?

**Comment by jschueller** [comment]
nevermind, maybe I thought there was some analytical method

**Comment by mbaudin47** [comment]
I have already managed to get some intermediate results, and I am currently working again on this topic. I will let you know when I have some results (the maths are there, the implementation is ongoing). In the meantime, please have a look at the following example:
https://github.com/mbaudin47/otsupgalilee-eleve/blob/master/4-Sensibilite/Exercice_SRC_with_linear_regression.ipynb
It shows how to use results from linear regression and apply them to squared SRC indices. At that time, I was working to get the _exact_ distribution of the importance factors. As an afterthought, I think that this might be a poor idea, because there always is a common expression in the numerator and the denominator, which is ignored in the script I used. I am currently working on the delta method. 
@regislebrun : This is related to your suggestion to work on Sobol' indices, since the structure of the expression is essentially the same.

**Comment by mbaudin47** [comment]
In order to implement this, I would rather create it as follows:
- Create a generic delta method.
- Implement the asymptotic distribution of the SRC indices would use the delta-method.

The generic delta method can be created from the `Distribution` of X, a `Function` and an integer. It provides the asymptotic distribution of g(sample mean(X)).

**Comment by mbaudin47** [comment]
This issue is managed within an ongoing contract.

**Comment by josephmure** [comment]
SRC confidence intervals can be computed analytically: https://www.jstatsoft.org/article/view/v020i08
The author provides an implementation in R: https://www.rdocumentation.org/packages/MBESS/versions/4.9.3/topics/ci.src
